src/generic/grid.cpp

```diff
@@ -4510,7 +4513,13 @@ wxGrid::DoGridCellDrag(wxMouseEvent& event,
     // Ctrl later can't change the dragging behaviour. Only the initial state
     // of the modifier keys matters.
     if ( m_selection )
+    {
+        if (isFirstDrag)
+        {
+            m_selection->StartSelecting();
```

Comment:
I wonder if we really need `StartSelecting()` at all, it looks like `ExtendCurrentBlock()` could (and IMO should) just set the internal flag to true automatically.

Reply:
I believe the "ExtendCurrentBlock" is used in other places (like keyboard selection) which may not be calling the EndSelecting() call to fire the SELECTED events.  I also wasn't sure what the sequences of events are if the developer actually calls the various SelectBlock or other selection methods.   In those cases, I assume it's just a SELECTED event and no SELECTING, but not really sure.  In my particular app, I only care about the case where the user is using a mouse to select.  All the other nuances are beyond what I care about at this point.